File: app/src/main/java/com/jakebarnby/filemanager/models/files/LocalFile.java
package com.jakebarnby.filemanager.models.files;

import android.net.Uri;

import com.jakebarnby.filemanager.util.Constants;

import java.io.File;

/**
 * Created by Jake on 6/5/2017.
 */

public class LocalFile extends SourceFile {

    public LocalFile() {
    }

    public void setFileProperties(File file) {
        setPath(file.getPath());
        setName(file.getName());
        setSourceName(Constants.Sources.LOCAL);
        setDirectory(file.isDirectory());
        setSize(file.length());
        setThumbnailLink(file.getAbsolutePath());
        setModifiedTime(file.lastModified());
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/models/sources/local/LocalFile.kt
package com.jakebarnby.filemanager.models.sources.local

import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.models.SourceType
import java.io.File

/**
 * Created by Jake on 6/5/2017.
 */
class LocalFile(file: File) : SourceFile() {

    init {
        path = file.path
        name = file.name
        id = file.absolutePath.hashCode().toLong()
        parentFileId = file.absolutePath.substring(0..file.absolutePath.lastIndexOf(File.separatorChar)).hashCode().toLong()
        sourceId = SourceType.LOCAL.id
        isDirectory = file.isDirectory
        isHidden = file.name.startsWith(".")
        size = file.length()
        thumbnailLink = file.absolutePath
        modifiedTime = file.lastModified()
    }
}